sources/packages/backend/libs/test-utils/src/factories/cas-supplier.ts

```diff
@@ -83,19 +83,22 @@ export function createFakeCASSupplier(
       siteProtected: "YES",
       lastUpdated: new Date(),
     };
-    casSupplier.supplierProtected = true;
+    casSupplier.supplierProtected =
+      options?.initialValues?.supplierProtected ?? true;
     casSupplier.isValid = false;
   }
-  casSupplier.supplierName = `${faker.name.lastName()}, ${faker.name.firstName()}`;
-  casSupplier.supplierNumber = faker.datatype
-    .number({ min: 100000, max: 9999999999 })
-    .toString();
+  casSupplier.supplierName =
+    options?.initialValues?.supplierName ??
+    `${faker.name.lastName()}, ${faker.name.firstName()}`;
+  casSupplier.supplierNumber =
+    options?.initialValues?.supplierNumber ??
+    faker.datatype.number({ min: 100000, max: 9999999999 }).toString();
```

Comment:
These new properties may be added if the `if` condition is added to the expected status.
When the record is created they will not be set.